src/Compilers/VisualBasic/Portable/Emit/SourceAssemblySymbolAdapter.vb

```diff
@@ -0,0 +1,29 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
+    Partial Friend Class SourceAssemblySymbol
+
+        Friend Function GetAssemblyCustomAttributesToEmit(compilationState As ModuleCompilationState,
+                                                          emittingRefAssembly As Boolean,
+                                                          emittingAssemblyAttributesInNetModule As Boolean) As IEnumerable(Of VisualBasicAttributeData)
+
+            Dim synthesized As ArrayBuilder(Of SynthesizedAttributeData) = Nothing
+            AddSynthesizedAttributes(compilationState, synthesized)
+
+            If emittingRefAssembly AndAlso Not HasReferenceAssemblyAttribute Then
+                Dim referenceAssemblyAttribute = Me.DeclaringCompilation?.
+                    TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_ReferenceAssemblyAttribute__ctor, isOptionalUse:=True)
+
+                Symbol.AddSynthesizedAttribute(synthesized, referenceAssemblyAttribute)
+            End If
+
+            ' PROTOTYPE(refout) How to avoid closure on emittingAssemblyAttributesInNetModule?
+            Return GetCustomAttributesToEmit(Me.GetAttributes(), synthesized,
+                                             Function(attribute, symbol)
+                                                 Return attribute.ShouldEmitAttribute(symbol, isReturnType:=False,
+                                                                                      emittingAssemblyAttributesInNetModule:=emittingAssemblyAttributesInNetModule)
```

Comment:
Note: I didn't manage to eliminate a closure here (no local functions in VB). Any tips?